src/activities/Elsa.Activities.Webhooks/ActivityTypes/WebhookActivityTypeProvider.cs

```
@@ -0,0 +1,99 @@
+        }
+
+        public async ValueTask<IEnumerable<ActivityType>> GetActivityTypesAsync(CancellationToken cancellationToken = default)
+        {
+            var specification = Specification<WebhookDefinition>.All;
+            var definitions = await _webhookDefinitionStore.FindManyAsync(specification, cancellationToken: cancellationToken);
+
+            var activityTypes = new List<ActivityType>();
+            foreach (var definition in definitions)
+            {
+                var activity = CreateWebhookActivityType(definition);
+                activityTypes.Add(activity);
+            }
+
+            return activityTypes;
+        }
+
+        private ActivityType CreateWebhookActivityType(WebhookDefinition webhook)
+        {
+            var typeName = webhook.Name;
+            var displayName = webhook.Name;
+
+            var descriptor = new ActivityDescriptor
+            {
+                Type = typeName,
+                DisplayName = displayName,
+                Category = WebhookActivityCategory,
+                Outcomes = new[] { OutcomeNames.Done },
+                Traits = ActivityTraits.Trigger,
+                Properties = new[]
+                {
+                    new ActivityPropertyDescriptor(
+                        "RequestMethod",
+                        "select",
+                        "Request Method",
+                        "Specify what request method this webhook should handle. Leave empty to handle both GET and POST requests",
+                        JObject.FromObject(new { Options = new[] { "", "GET", "POST" } }))
+                }
+            };
+
+            async ValueTask<IActivity> ActivateActivityAsync(ActivityExecutionContext context, WebhookDefinition webhook)
+            {
+                var activity = await _activityActivator.ActivateActivityAsync(context, typeof(HttpEndpoint));
+
+                if (activity is HttpEndpoint httpEndpointActivity)
```

It appears to me that this check is unnecessary - `_activityActivator.ActivateActivityAsync(context, typeof(HttpEndpoint))` will always return an activity of type `HttpEndpoint` (unless there's a bug, in which case we want to fail hard).